CHANGELOG.md

```diff
@@ -1,17 +1,44 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
 ## Unreleased
 
-## 2018-01-12 - Unsupported Release 0.1.0
+### Added
+
+- `compatibility` feature and flag (usable only with the `taskscheduler_api2` provider), allowing users to specify which version of Scheduled Tasks the task should be compatible with ([MODULES-6526](https://tickets.puppetlabs.com/browse/MODULES-6526))
+- Documentation of both the legacy `win32_taskscheduler` provider and the new `taskscheduler_api2` into the project readme, ensuring users will not need to refer to Puppet core documentation ([MODULES-6417](https://tickets.puppetlabs.com/browse/MODULES-6417))
+- New helper for `taskscheduler_api2` allowing it to manage scheduled tasks of any compatibility level ([MODULES-6844](https://tickets.puppetlabs.com/browse/MODULES-6844), [MODULES-6845](https://tickets.puppetlabs.com/browse/MODULES-6845))
+
+### Changed
 
-### Summary
+- Default provider from `win32_taskscheduler` to `taskscheduler_api2` ([MODULES-6591](https://tickets.puppetlabs.com/browse/MODULES-6591))
+- Logic for managing triggers, refactoring for improved maintainability ([MODULES-6843](https://tickets.puppetlabs.com/browse/MODULES-6843), [MODULES-6895](https://tickets.puppetlabs.com/browse/MODULES-6895))
```

Comment:
@Iristyle, @clairecadman, @glennsarti: Not sure if we should keep these ticketed entries describing changes to internals in the changelog. They're significant but mostly internal-facing.

Reply:
Specifically unsure of lines 12, 17, and 18.

Reply:
Yeah, I'm fine with it... It's highly unlikely anyone is relying on this in any way, but I'm OK with it being in here for the sake of anyone working on the project.